cmd/config/config.go

```diff
@@ -54,7 +55,31 @@ func run() func(*cobra.Command, []string) error {
 
 			fmt.Fprint(cmd.OutOrStdout(), string(configJSON)+"\n")
 		case viper.GetBool(SetFlag):
-			fmt.Fprintln(cmd.OutOrStdout(), "called --set flag")
+			// flag needs two arguments: a key and value
+			if len(args)%2 != 0 {
+				return errors.New("flag needs an argument: --set")
+			}
+
+			rawConfig, v, err := getRawConfig()
+			if err != nil {
+				return err
+			}
+
+			// add arg pairs to config
+			// where each argument is --set arg1 val1 --set arg2 val2
+			for i, a := range args {
+				if i%2 == 0 {
+					rawConfig[a] = struct{}{}
+				} else {
+					rawConfig[args[i-1]] = a
+				}
+			}
+
+			setKeyFn := func(key string, value interface{}, v *viper.Viper) {
```

Comment:
Using a filter function here will be more obvious in the next PR that supports `--unset`.